### ISSUE TITLE
Multi factor enrollment failing with internal error if `displayName` is `nil`

### ISSUE THREAD

**Issue by alexandersandberg** [firebase/firebase-ios-sdk#13856]
### Description

When [enrolling multi factor](https://firebase.google.com/docs/auth/ios/multi-factor#enrolling_a_second_factor), you call the following enroll method:

```swift
func enroll(
    with assertion: MultiFactorAssertion,
    displayName: String?
) async throws
```

The `displayName` property here is optional.

However, we've noticed after upgrading from 10.24.0 to 11.3.0 that the call errors with an internal error stating "Request contains an invalid argument" if `displayName` is `nil`.

If we set `displayName` to e.g. an empty string `""` instead, it works as expected.

### Reproducing the issue

```swift
let verificationId = ...
let verificationCode = ...

let credential = PhoneAuthProvider.provider().credential(withVerificationID: verificationId, verificationCode: verificationCode)
let assertion = PhoneMultiFactorGenerator.assertion(with: credential)

try await Auth.auth().currentUser?.multiFactor.enroll(with: assertion, displayName: nil) // (1)
try await Auth.auth().currentUser?.multiFactor.enroll(with: assertion, displayName: "") // (2)
```

- (1, `displayName: nil`) fails with internal error "Request contains an invalid argument."
- (2, `displayName: ""`) works as expected

### Firebase SDK Version

11.3.0

### Xcode Version

16.0

### Installation Method

Swift Package Manager

### Firebase Product(s)

Authentication

### Targeted Platforms

iOS

### Relevant Log Output

```shell
Error Domain=FIRAuthErrorDomain Code=17999 "An internal error has occurred, print and inspect the error details for more information." UserInfo={FIRAuthErrorUserInfoNameKey=ERROR_INTERNAL_ERROR, NSUnderlyingError=0x3005ac4e0 {Error Domain=FIRAuthInternalErrorDomain Code=3 "(null)" UserInfo={FIRAuthErrorUserInfoDeserializedResponseKey={
    code = 400;
    message = "Request contains an invalid argument.";
    status = "INVALID_ARGUMENT";
}, NSUnderlyingError=0x300594270 {Error Domain=com.google.HTTPStatus Code=400 "(null)" UserInfo={data={length = 127, bytes = 0x7b0a2020 22657272 6f72223a 207b0a20 ... 220a2020 7d0a7d0a }, data_content_type=application/json; charset=UTF-8}}}}, NSLocalizedDescription=An internal error has occurred, print and inspect the error details for more information.}
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by ncooke3** [comment]
Thanks for the report, @alexandersandberg. This was indeed a regression, and will be fixed in the next Firebase release, 11.4.

**Comment by alexandersandberg** [comment]
Thank you for fixing this so quickly! 🙏
